Sources/SwiftMemcache/MemcachedResponseDecoder.swift

```diff
@@ -151,6 +166,28 @@ struct MemcachedResponseDecoder: NIOSingleStepByteToMessageDecoder {
             let response = MemcachedResponse(returnCode: returnCode, dataLength: dataLength)
             self.nextStep = .returnCode
             return .returnDecodedResponse(response)
+
+        case .decodeValue(let returnCode, let dataLength, let flags):
+            guard buffer.readableBytes >= dataLength else {
```

Comment:
We should probably check here for `dataLength` + carriageReturn + newLine. To make sure we are not decoding the data until we also have the returns